File: ruby/lotto_picker/test_lpickr.rb
#! /usr/bin/env ruby
require 'minitest/autorun'
require './lpickr'

class LottoPickerTest < Minitest::Test
  def test_returns_expected_number_of_numbers
    assert_equal 1, pick(1, 40).length
    assert_equal 6, pick(6, 40).length
    assert_equal 12, pick(12, 40).length
  end

  def test_returns_all_numbers_within_range
    100.times do
      result = pick(1, 10)
      assert_equal 1, result.length
      assert result.first.between?(1, 10)
    end
  end

  def test_fails_if_ask_for_more_numbers_than_available
    assert_raises(StandardError) do
      pick(6, 5)
    end
  end

  def test_empty_if_ask_for_zero
    assert_equal 0, pick(0, 0).length
  end
end

File: python/lotto_picker/test_lpickr.py
#! /usr/bin/env python
#
import unittest
from lpickr import pick


class LpickrTests(unittest.TestCase):

    def test_returns_expected_number_of_numbers(self):
        self.assertEqual(1, len(pick(1, 40)))
        self.assertEqual(6, len(pick(6, 40)))
        self.assertEqual(12, len(pick(12, 40)))

    def test_returns_all_numbers_within_range(self):
        for _ in range(100):
            result = pick(1, 10)
            self.assertEqual(1, len(result))
            self.assertTrue(1 <= result[0] <= 10)

    def test_fails_if_ask_for_more_numbers_than_available(self):
        self.assertRaises(Exception, lambda: pick(6, 5))

    def test_empty_if_ask_for_zero(self):
        self.assertEqual(0, len(pick(0, 0)))


if __name__ == '__main__':
    unittest.main()
